base/linalg/triangular.jl

```diff
@@ -1757,19 +1757,10 @@ function logm(A0::UpperTriangular{T}) where T<:Union{Float64,Complex{Float64}}
     m = 0
 
     # Compute repeated roots
-    d = diag(A)
-    dm1 = Vector{T}(n)
+    d = complex(diag(A))
     s = 0
-    for i = 1:n
-        dm1[i] = d[i] - 1.
-    end
-    while norm(dm1, Inf) > theta[tmax]
-        for i = 1:n
-            d[i] = sqrt(d[i])
-        end
-        for i = 1:n
-            dm1[i] = d[i] - 1
-        end
+    while norm(d-1, Inf) > theta[tmax] && s < maxsqrt
```

Comment:
`dm1` changes again inside the while loop. I think both approaches are equal in time and memory, this one is just shorter. No?